server/cherry_pick.go

```diff
@@ -55,7 +55,10 @@ func (s *Server) handleCherryPick(ctx context.Context, eventIssueComment IssueCo
 	}
 }
 
-func (s *Server) checkIfNeedCherryPick(ctx context.Context, pr *model.PullRequest) {
+func (s *Server) checkIfNeedCherryPick(pr *model.PullRequest) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultRequestTimeout*time.Second)
```

Comment:
Maybe a couple of sentences that explains why we are creating context here would be useful for future refactors.